### ISSUE TITLE
Cannot use WebIO in IJulia inside a folder called `notebooks`

### ISSUE THREAD

**Issue by rdeits** [JuliaGizmos/WebIO.jl#120]
Yeah, this one was a bit of a stumper. 

If you create a folder called `notebooks`, start `jupyter notebook` outside that folder, and then try to create a notebook inside the `notebooks` folder, then using WebIO will fail. This happens because the regex to detect hosts which use a `notebooks` prefix, isn't quite right. When inside a folder literally called `notebooks`, the URL is: 

```
http://localhost:8888/notebooks/notebooks/foo.ipynb 
```

and the regex is: `/(.*)\\/notebooks\\/.*\\.ipynb/`. Since regexes are greedy, rather than matching `http://localhost:8888/`, the regex keeps going and ends up matching `http://localhost:8888/notebooks/`. That results in the wrong base URL, so all the WebIO pkg URLs are broken.

Basically: 

![](https://imgs.xkcd.com/comics/perl_problems.png)

Anyway, I'll see if I can fix this. 

@tkoolen, by the way, this is why the new MeshCat didn't work the first time I tried to show you (because we were trying out one of your demos which happened to live in a folder called `notebooks`).

**Comment by rdeits** [comment]
Also, the word "notebooks" has now lost all meaning for me.

**Comment by shashi** [comment]
This is bad haha!

**Comment by shashi** [comment]
Thanks for fixing this monstrosity.
